File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/StudyRoom.java
package de.tum.in.tumcampusapp.models.tumcabe;

import java.util.Date;

/**
 * Representation of a study room.
 */
public class StudyRoom {

    public final int id;
    public String code = "";
    public String name = "";
    public String location = "";
    public final Date occupiedTill;

    public StudyRoom(int id, String code, String name, String location, Date occupiedTill) {
        this.id = id;
        this.code = code;
        this.name = name;
        this.location = location;
        this.occupiedTill = occupiedTill;
    }

    @Override
    public String toString() {
        return "" + code;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/StudyRoom.kt
package de.tum.`in`.tumcampusapp.models.tumcabe

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import java.util.*

/**
 * Representation of a study room.
 */
@Entity(tableName = "study_rooms")
data class StudyRoom(@field:PrimaryKey
                     var id: Int = -1,
                     var code: String = "",
                     var name: String = "",
                     var location: String = "",
                     @ColumnInfo(name = "group_id")
                     var studyRoomGroup: Int = -1,
                     @ColumnInfo(name = "occupied_till")
                     var occupiedTill: Date = Date()) {
    override fun toString() = code
}
